### PR TITLE
Add scripts for ResNet in Fleet/benchmark/collective

### PR DESCRIPTION
**Pull request by sandyhouse**
Add scripts for ResNet in the directory Fleet/benchmark/collective.

### CODE REVIEW

File: benchmark/collective/resnet/launch.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+def main():
+    args = parse_args()
+
+    # launch multiple training process
+    start_procs(args.gpus, args.entrypoint_script, args.entrypoint_args,
+                args.log_dir)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (gavin1332) [inline benchmark/collective/resnet/launch.py]: use paddle.distributed.launch

Comment (sandyhouse) [reply benchmark/collective/resnet/launch.py]: Done

File: benchmark/collective/resnet/scripts/train_gpu_mp.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export NCCL_DEBUG=INFO
+
+# Change your training arguments:
+python -m paddle.distributed.launch ./train_with_pe_fleet.py \
+       --data_dir /ssd2/lilong/ImageNet --start_test_pass 0
+
+# For paddlecloud
+# python -m paddle.distributed.launch --cluster_node_ips=$PADDLE_TRAINERS \
+#     --node_ip=$POD_IP ./train_with_pe_fleet.py \
+#     --data_dir /ssd2/lilong/ImageNet --start_test_pass 0
```
Comment (gavin1332) [inline benchmark/collective/resnet/scripts/train_gpu_mp.sh]: controlled by shell argument with ifelse statement but not by comment

Comment (sandyhouse) [reply benchmark/collective/resnet/scripts/train_gpu_mp.sh]: Done

**Comment by sandyhouse** [comment]
Todo List for best performance:
- [ ] be compatible with paddlepaddle development branch
- [ ] add the configuration for num_comm

**Pull request merged**